src/sentry/api/endpoints/organization_details.py

```diff
@@ -954,7 +967,45 @@ def put(self, request: Request, organization) -> Response:
             with transaction.atomic(router.db_for_write(Organization)):
                 organization, changed_data = serializer.save()
 
-            if "targetSampleRate" in changed_data:
+            if (
+                organization.get_option("sentry:sampling_mode")
+                == DynamicSamplingMode.ORGANIZATION.value
+            ) and "targetSampleRate" in changed_data:
+                with transaction.atomic(router.db_for_write(ProjectOption)):
+                    organization.update_option(
+                        "sentry:target_sample_rate", request.data["targetSampleRate"]
+                    )
```

Comment:
Is this needed? We are doing this later on line 1000?

Reply:
When the targetSampleRate is changed, but the samplingMode is not, only the targetSampleRate should be set.
When the targetSampleRate is changed, and the samplingMode is also set and changed to `organization`, the mode should be changed, the targetSampleRate should be set, and the project option sentry:sampling_rate should be deleted. So we do need the branch, but I'll refactor in a simpler way.

Reply:
Yep, sorry, my bad, miss-read the key that we check for